### PR TITLE
Closes #1191: Update DNet releases repositories

### LINKED ISSUE #1191: Update DNet releases repositories
Following information from CNR we need to update maven repositories locations. Current repository URL `http://maven.research-infrastructures.eu` must be replaced with new one `https://maven.d4science.org/nexus`. This change is probably responsible for following warnings from integration tests log:

`
[main] WARN org.codehaus.plexus.PlexusContainer - Failure to transfer dom4j:dom4j/maven-metadata.xml from http://maven.research-infrastructures.eu/nexus/content/repositories/dnet-deps was cached in the local repository, resolution will not be reattempted until the update interval of dnet-deps has elapsed or updates are forced. Original error: Could not transfer metadata dom4j:dom4j/maven-metadata.xml from/to dnet-deps (http://maven.research-infrastructures.eu/nexus/content/repositories/dnet-deps): Connect to maven.research-infrastructures.eu:80 [maven.research-infrastructures.eu/146.48.122.71] failed: Connection timed out  \r
`

### PR DESCRIPTION
**Pull request by przemyslawjacewicz**

### CODE REVIEW

**Review by marekhorst — APPROVED** [review]
LGTM.

**Comment by przemyslawjacewicz** [comment]
Unfortunately the build timed out after 1h. Simple replacement of DNet repositories in IIS POM was not enough because old repositories are still used in DNet dependencies and the repositories are queried during building. The query times out but it _takes time_ and as a result the time of the build exceeds threshold and the build is killed. The query is triggered probably by the fact that we're using ranges in dependency version specification. The simplest solution I've found is to not replace old repositories with new ones but to explicitly disable old repositories and add new repositories. I've committed this change to this PR.

**Comment by przemyslawjacewicz** [comment]
One additional remark: after merging #1166 Jenkins build time substantially increased. This is due to the fact that some long local tests that were previously marked as integration tests and were not executed during maven `test` phase are now marked as slow tests and are executed during `test` phase. One example is  `eu.dnetlib.iis.wf.affmatching.match.voter.AffOrgMatchVoterStrengthEstimatorAndTest` which takes more than 900s.

**Comment by przemyslawjacewicz** [comment]
Re-requesting review due to additional changes.

File: pom.xml
Comment (marekhorst) [inline pom.xml:62]: OK, so we are blacklisting it explicitly here in order to make sure D-Net transitive dependencies are not retrieved from an old repo, right?

So it is a temporary solution until the repo is resolved in DNet part of the project we are relying on.

I would introduce just a minor change: since the new repo is the one which should be used I would drop the `-new` suffix (indicating this repo as the main one) and add an `-old` suffix (or `-deprecated`?, something indicating this is obsolete) to the old repo to indicate this is the one to be removed. So the 2nd phase of the repo transition will be simple `dnet45-releases-old` repo removal from the `pom.xml` file.

**Review by marekhorst — APPROVED** [review]
Just one recommendation in a comment.

File: pom.xml
Comment (przemyslawjacewicz) [reply pom.xml]: Due to updates this part of POM was changed and no new/old/deprecated repos are needed. I'm commenting on the solution in the main conversation.

**Comment by przemyslawjacewicz** [comment]
It turns out I've made a mistake and we do not need to explicitly disable old/deprecated maven repositories. The repository that is problematic is referenced through IIS dependencies but it is not an IIS repository. It is located at `http://maven.research-infrastructures.eu/nexus/content/repositories/dnet-deps/` and is a repository listed for example in `dnet45-parent`. I've added this repository with new url to IIS repositories and Jenkins build run without any timeouts.

**Review by marekhorst — APPROVED** [review]
I guess it's OK. Does it mean we should still understand `dnet-deps` it as a temporary workaround? I mean we do not relate on this repo in IIS, it is a transitive repository coming from DNet. If this is the case then we should probably mark it somehow (e.g. as an issue) to drop it once the original repository defined in `dnet45-parent` is fixed.

**Comment by claudioatzori** [comment]
Well, all the references in the dnet-parent modules (both snapshot and releases) were hotfixed to use the new maven repository base URL, e.g.
* [dnet45-bootstrap-release ](https://maven.d4science.org/nexus/service/local/repositories/dnet45-bootstrap-release/content/eu/dnetlib/dnet45-parent/1.0.0/dnet45-parent-1.0.0.pom)
* [ dnet45-bootstrap-snapshot](https://maven.d4science.org/nexus/service/local/repositories/dnet45-bootstrap-snapshot/content/eu/dnetlib/dnet45-parent/1.0.0-SNAPSHOT/dnet45-parent-1.0.0-20201113.093220-5.pom)

So once all the local maven repos (on jenkins, workstations) would fetch the updated parents I would have expected to not see anymore any resolution attempt on the old URL... however I still see some references in the builds and I still haven't figured out the cause. Perhaps sone of the submodules is still including the old URL in its pom. I'll let you know as soon as I have news.

**Comment by claudioatzori** [comment]
Scanning the codebase on my checkout highlighted the following modules still including references to the deprecated URLs

```⋊> ~/w/dnet45 ag 'http://maven.research-infrastructures.eu/' */pom.xml                                                                                                                                                                    16:49:42
dnet-actionmanager-common/pom.xml
25:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-central</url>
30:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-snapshots</url>

dnet-dedup/pom.xml
44:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet45-releases</url>
53:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet-deps</url>
59:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet45-releases</url>
68:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet45-snapshots</url>

dnet-hadoop-commons/pom.xml
25:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-central</url>
30:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-snapshots</url>
35:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/typesafe</url>

dnet-hadoop-service/pom.xml
24:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-central</url>
29:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-snapshots</url>

dnet-mapreduce-jobs-TRUNK/pom.xml
42:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-central</url>
47:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/cloudera-snapshots</url>

dnet-modular-index-service/pom.xml
19:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/typesafe</url>

dnet-openaire-data-protos-IIS-CDH-5.3.0/pom.xml
26:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet4-bootstrap-release/</url>

dnet-openaire-data-protos/pom.xml
26:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet4-bootstrap-release/</url>

dnet-openaire-exporter/pom.xml
20:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet45-releases</url>
36:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet-deps</url>
42:			<url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet45-releases</url>
51:            <url>http://maven.research-infrastructures.eu/nexus/content/repositories/dnet45-snapshots</url>```

I already prepared a new release for all the modules listed above, replacing the base URL with the new one, except for *dnet-dedup*, which will be updated soon by Michele De Bonis.

**Comment by przemyslawjacewicz** [comment]
Ok, great, for the time being I think we can merge this PR and remove the repository later. I've created an issue for this #1194 .

**Comment by marekhorst** [comment]
> Ok, great, for the time being I think we can merge this PR and remove the repository later. I've created an issue for this #1194 .

Great, thank you Przemek.

**Pull request merged**
